VERSIONING.md

```diff
@@ -0,0 +1,119 @@
+# Versioning
+
+This document aims to provide an overview of how VAST is versioned, and what
+significance a version bump carries.
+
```

Comment:
Here, I'm missing a quick overview about the various pieces that we are versioning in VAST.